types/request/index.d.ts

```diff
@@ -197,6 +197,10 @@ declare namespace request {
         };
     }
 
+    interface PoolOptions {
+        maxSockets?: number;
```

Comment:
Instead of having the two long one-liners, does make more sense to move the `[key: string]: http.Agent | https.Agent` into this interface?

Reply:
Your suggest is does not work, because `maxSockets` consider with `[key: string]` and compiler check to it case.

Reply:
Yep, you're right.

What about:
```ts

interface ExtraPoolOptions {
  maxSockets?: number;
}

type PoolOptions = false | { [key: string]: http.Agent | https.Agent } & ExtraPoolOptions | ExtraPoolOptions;
```
I'd still like to reduce the two duplicate lines if possible.

Reply:
Thanks for your help